assets/src/edit-story/components/richText/karma/multiSelection.karma.js

```diff
@@ -0,0 +1,254 @@
+      const {
+        bold,
+        italic,
+        underline,
+        fontWeight,
+        letterSpacing,
+        fontColor,
+      } = data.fixture.editor.inspector.designPanel.textStyle;
+
+      // Select both text fields
+      await selectBothTextFields();
+
+      // Check all styles are default
+      expect(bold.checked).toBe(false);
+      expect(italic.checked).toBe(false);
+      expect(underline.checked).toBe(false);
+      expect(fontWeight.value).toBe('Regular');
+      expect(letterSpacing.value).toBe('0%');
+      expect(fontColor.hex.value).toBe('000000');
+
+      // Toggle all styles
+      await data.fixture.events.click(italic.button);
+      await data.fixture.events.click(underline.button);
+      await data.fixture.events.click(fontWeight.select);
+      await data.fixture.events.click(fontWeight.option('Black'));
+      await data.fixture.events.click(letterSpacing, { clickCount: 3 });
+      await data.fixture.events.keyboard.type('50');
+      await data.fixture.events.keyboard.press('Escape');
+      await data.fixture.events.click(fontColor.hex, { clickCount: 3 });
+      await data.fixture.events.keyboard.type('FF00FF');
+      await data.fixture.events.keyboard.press('Escape');
+
+      // Verify all styles, now expected to be updated
+      expect(bold.checked).toBe(true);
+      expect(italic.checked).toBe(true);
+      expect(underline.checked).toBe(true);
+      expect(fontWeight.value).toBe('Black');
+      expect(letterSpacing.value).toBe('50%');
+      expect(fontColor.hex.value).toBe('FF00FF');
+
+      // Assume text contents to match expectation
+      const css = [
+        'font-weight: 900',
+        'font-style: italic',
+        'text-decoration: underline',
+        'color: #f0f',
+        'letter-spacing: 0.5em',
+      ].join('; ');
+      const getExpected = (content) => `<span style="${css}">${content}</span>`;
+      expect(getTextContent(0)).toBe(getExpected('Fill in some text'));
+      expect(getTextContent(1)).toBe(getExpected('Number #2'));
+    });
+
+    it('should apply formatting correctly for multi-style text field', async () => {
+      const {
+        bold,
+        italic,
+        underline,
+        fontWeight,
+        letterSpacing,
+        fontColor,
+      } = data.fixture.editor.inspector.designPanel.textStyle;
+
+      // Edit formatting for second text field
+      await data.fixture.events.click(letterSpacing, { clickCount: 3 });
+      await data.fixture.events.keyboard.type('50');
+      await data.fixture.events.keyboard.press('Escape');
+      await data.fixture.events.click(fontColor.hex, { clickCount: 3 });
+      await data.fixture.events.keyboard.type('FF00FF');
+      await data.fixture.events.keyboard.press('Escape');
+      await data.fixture.events.click(italic.button);
+      await data.fixture.events.click(underline.button);
+      await data.fixture.events.click(fontWeight.select);
+      await data.fixture.events.click(fontWeight.option('Black'));
```

Comment:
Again, it would be nice to test if the font families differ between the two text elements. Maybe the line heights as well? Not sure if it's tested somewhere else?

Reply:
These are not inline styles, so not a focus of this test.